packages/contracts/contracts/optimistic-ethereum/utils/libraries/GasConsumer.sol

```diff
@@ -1,19 +1,43 @@
 pragma solidity ^0.5.0;
 
 contract GasConsumer {
-
-    // default fallback consumes all allotted gas
+    // default fallback--consumes all allotted gas
     function() external {
-        uint i;
-        while (true) {
-            i += 420;
+        assembly {
+            invalid()
+        }
+    }
+
+    // Overhead for checking methodId etc in this function before the actual call()
+    // This was figured out empirically during testing--adding methods or changing compiler settings will require recalibration.
+    uint constant constantOverheadEOA = 947;
```

Comment:
Will this ever change through hardforks, and if so, how can we update it?

Reply:
It could, but it's a case where other critical components will break first.  In this particular instance I believe the impact would be a constant size change in OVM gas consumed, but still deterministic.